lib/cc/engine/duplication.rb

```diff
@@ -21,13 +22,15 @@ def run
         languages.each do |lang|
           next unless language_supported?(lang)
```

Comment:
thoughts on, instead of languages, storing a hash of language to class, or just an array of the supported analyzers, to avoid some of the class-oriented programming?